doc/extdev/appapi.rst

```diff
@@ -23,387 +23,73 @@ These methods are usually called in an extension's ``setup()`` function.
-
-   Add the standard docutils :class:`Transform` subclass *transform* to the list
-   of transforms that are applied after Sphinx parses a reST document.
-
-.. method:: Sphinx.add_post_transform(transform)
-
-   Add the standard docutils :class:`Transform` subclass *transform* to the list
-   of transforms that are applied before Sphinx writes a document.
-
-.. method:: Sphinx.add_javascript(filename)
-
-   Add *filename* to the list of JavaScript files that the default HTML template
-   will include.  The filename must be relative to the HTML static path, see
-   :confval:`the docs for the config value <html_static_path>`.  A full URI with
-   scheme, like ``http://example.org/foo.js``, is also supported.
-
-   .. versionadded:: 0.5
-
-.. method:: Sphinx.add_stylesheet(filename, alternate=None, title=None)
-
-   Add *filename* to the list of CSS files that the default HTML template will
-   include.  Like for :meth:`add_javascript`, the filename must be relative to
-   the HTML static path, or a full URI with scheme.
-
-   .. versionadded:: 1.0
-
-   .. versionchanged:: 1.6
-      Optional ``alternate`` and/or ``title`` attributes can be supplied with
-      the *alternate* (of boolean type) and *title* (a string) arguments. The
-      default is no title and *alternate* = ``False`` (see `this explanation
-      <https://developer.mozilla.org/en-US/docs/Web/CSS/Alternative_style_sheets>`_).
-
-.. method:: Sphinx.add_latex_package(packagename, options=None)
-
-   Add *packagename* to the list of packages that LaTeX source code will include.
-   If you provide *options*, it will be taken to `\usepackage` declaration.
+.. currentmodule:: sphinx.application
 
-   .. code-block:: python
+.. automethod:: Sphinx.setup_extension(name)
 
-      app.add_latex_package('mypackage')             # => \usepackage{mypackage}
-      app.add_latex_package('mypackage', 'foo,bar')  # => \usepackage[foo,bar]{mypackage}
+.. automethod:: Sphinx.add_builder(builder)
 
-   .. versionadded:: 1.3
+.. automethod:: Sphinx.add_config_value(name, default, rebuild)
 
-.. method:: Sphinx.add_lexer(alias, lexer)
+.. automethod:: Sphinx.add_domain(domain)
 
-   Use *lexer*, which must be an instance of a Pygments lexer class, to
-   highlight code blocks with the given language *alias*.
+.. automethod:: Sphinx.override_domain(domain)
 
-   .. versionadded:: 0.6
+.. automethod:: Sphinx.add_index_to_domain(domain, index)
 
-.. method:: Sphinx.add_autodocumenter(cls)
+.. automethod:: Sphinx.add_event(name)
 
-   Add *cls* as a new documenter class for the :mod:`sphinx.ext.autodoc`
-   extension.  It must be a subclass of :class:`sphinx.ext.autodoc.Documenter`.
-   This allows to auto-document new types of objects.  See the source of the
-   autodoc module for examples on how to subclass :class:`Documenter`.
+.. automethod:: Sphinx.set_translator(name, translator_class)
 
-   .. XXX add real docs for Documenter and subclassing
+.. automethod:: Sphinx.add_node(node, \*\*kwds)
```

Comment:
Is this escape needed?

Reply:
It is needed in order to enable proper syntax highlighting when editing the docs. It has no impact on the output.